pkgs/desktops/mate/python-caja/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, pkgconfig, intltool, gtk3, caja, pythonPackages }:
+
+stdenv.mkDerivation rec {
+  name = "python-caja-${version}";
+  version = "${major-ver}.${minor-ver}";
+  major-ver = "1.18";
+  minor-ver = "1";
+
+  src = fetchurl {
+    url = "http://pub.mate-desktop.org/releases/${major-ver}/${name}.tar.xz";
+    sha256 = "0n43cvvv29gq31hgrsf9al184cr87c3hzskrh2593rid52kwyz44";
+  };
+
+  nativeBuildInputs = [
+    pkgconfig
+    intltool
+    pythonPackages.wrapPython
+  ];
+
+  buildInputs = [
+    gtk3
+    caja
+    pythonPackages.python
+    pythonPackages.pygobject3
+  ];
+
+  configureFlags = [ "--with-cajadir=$$out/lib/caja/extensions-2.0" ];
```

Comment:
Does this actually eventualy get expanded correctly? Usually it's done like this:
````
preConfigure = ''
    configureFlagsArray+=("--with-cajadir=$$out/lib/caja/extensions-2.0")
'';
````

Reply:
It is working here and in other packages too.

```
$ nix-build --option build-use-chroot true -K -A mate.python-caja 2>&1 | tee /var/tmp/nix-build.log
[...]
Libraries have been installed in:
   /nix/store/72q7hql2lwwmgqm5jd3lr896mqkv53fq-python-caja-1.18.1/lib/caja/extensions-2.0
[...]
```

Reply:
Searching for `'$$'` in nix files shows several other cases of using `'$$'` in `configureFlags`.

Reply:
Ok, fair enough. But it really is a bad pattern to do, it's a completely unpredictable whether it's going to work or not, or even fail silently.

For example before commit 3d59a8e1a8341d09689d4ace8765ae134c264fe7 ncurses appeared to build fine, but the pkgconfig file was wrong, containing unexpanded `$(out)` / `$$out` etc.

Reply:
In the next updates I can abandon this bad pattern and follow your suggestion.